app/helpers/application_helper.rb

```diff
@@ -123,4 +123,12 @@ def user_mem_link(user_id, username, deleted)
     return '(deleted user)'.html_safe if deleted
     link_to username, user_path(user_id)
   end
+
+  def conditional_tag(tag_name, condition, &block)
+    if condition
+      content_tag tag_name, capture(&block)
+    else
+      capture(&block)
+    end
+  end
```

Comment:
...this looks like *black magic*. How does it *work*.

(Also probably it should be in post_helper?)

Reply:
So I think I actually grabbed the original setup from the internet (not sure where exactly, but e.g. https://stackoverflow.com/questions/5809917/conditional-tag-wrapping-in-rails-erb#5810148 uses `capture` too)

[`capture`](https://api.rubyonrails.org/classes/ActionView/Helpers/CaptureHelper.html#method-i-capture) takes the result of the block (which is how tag wrapping works in HAML, when you do things like `link_to <url> do`, the `do` bit is indicating a block, and then you have stuff nested inside the link on the next line)

If the condition is true, it takes the capture result and passes it to `content_tag tag_name` ([`content_tag`](https://apidock.com/rails/ActionView/Helpers/TagHelper/content_tag) takes content as its second parameter, or you can pass a block) otherwise it just returns the value of the block itself, which is how these wrapping setups usually seem to be done in helpers!

This could maybe be in post_helper! I'm not sure, I figured it could stay in application_helper since it's quite a generic helper and could be used elsewhere, but since we only use it in posts views right now, we could move it there and bring it back to application_helper only if/when we do use it elsewhere?

Reply:
...I do not feel much less like I am looking at black magic. The only part of that I understood was 'this involves blocks somehow'. and that it accepts a generic condition to check if it uses the tag its given or just the block.

...kind of feel like the genericization may be making this more confusing than it needs to be?